src/audio/drc/drc_math_hifi3.c

```diff
@@ -0,0 +1,263 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Google LLC. All rights reserved.
+//
+// Author: Pin-chih Lin <johnylin@google.com>
+
+#include <sof/audio/drc/drc_math.h>
+#include <sof/math/numbers.h>
+
+#if DRC_HIFI3
+
+#include <xtensa/tie/xt_hifi3.h>
+
+/*
+ * A substitutive function of Q_MULTSR_32X32(a, b, qa, qb, qy) in HIFI manners.
+ *
+ * In AE_MULF32R, it takes the multiplication of 1.31 and 1.31 to 17.47, which means it does
+ * right-shift 15 bits. Let's consider a is .qa, b is .qb, then tmp will be .(qa+qb-15)
+ * In AE_ROUND32F48SSYM, it rounds 17.47 to 1.31, equally right-shifts 16 bits. Let's consider the
+ * output is .qy, then tmp here needs to be .(qy+16)
+ * If we set the left-shift bit from the former tmp to the latter tmp as "x", then:
+ *     (qa+qb-15) + x = (qy+16)
+ *     x = qy-qa-qb+31
+ */
+static inline ae_f32 mult(ae_f32 a, ae_f32 b, int32_t qa, int32_t qb, int32_t qy)
+{
+	ae_f64 tmp;
+
+	tmp = AE_MULF32R_LL(a, b);
+	tmp = AE_SLAA64S(tmp, qy - qa - qb + 31);
+	return AE_ROUND32F48SSYM(tmp);
+}
+
+/*
+ * Input depends on precision_x
+ * Output range [0.5, 1); regulated to Q2.30
+ */
+static inline ae_f32 rexp_fixed(ae_f32 x, int32_t precision_x, int32_t *e)
+{
+	int32_t bit = 31 - norm_int32((int32_t)x);
```

Comment:
I think there's an intrinsic to get # of shifts to normalize. AE_NSAZ32.L perhaps?